libcudacxx/include/cuda/std/__cccl/compiler.h

```diff
@@ -42,11 +42,10 @@
 
 // Determine the host compiler and its version
 #if defined(__INTEL_COMPILER)
-#  define _CCCL_COMPILER_ICC 1
-#  ifndef CCCL_SUPPRESS_ICC_DEPRECATION_WARNING
-#    warning \
-      "Support for the Intel C++ Compiler Classic is deprecated and will eventually be removed. Define CCCL_SUPPRESS_ICC_DEPRECATION_WARNING to suppress this warning"
-#  endif // CCCL_SUPPRESS_ICC_DEPRECATION_WARNING
+#  ifndef CCCL_SUPPRESS_UNSUPPORED_COMPILER_ERROR
+#    error \
+      "The Intel C++ Compiler Classic (icc/icpc) is not supported by CCCL. Define CCCL_SUPPRESS_UNSUPPORED_COMPILER_ERROR to suppress this error."
+#  endif // !CCCL_SUPPRESS_UNSUPPORED_COMPILER_ERROR
```

Comment:
I replaced the deprecation warning by a hard error, which is suppressible by defining `CCCL_SUPPRESS_UNSUPPORED_COMPILER_ERROR`.